docs/Building.md

```diff
@@ -1,28 +1,173 @@
+
+### Running the system tests
+
+The Calico Docker system tests provide detailed testing of the calicoctl command line tool and the calio/node
+container image.  The tests cover multiple topologies and covers most of the available calicoctl command line
+options.
+
+The full suite of system tests takes approximately 30 minutes to complete.  If you are developing code for Calico
+Docker, and are regularly running the system tests, a faster subset of system tests may be run which covers most
+mainline function.
+
+Both ST suites utilize Docker's [libnetwork network driver](https://github.com/docker/libnetwork), available in the
+Docker [experimental channel](https://github.com/docker/docker/tree/master/experimental) alongside the Docker 1.7
+release.
+
+Both STs start a consul server and etcd server bound to a local IP address (automatically determined).  If you have
+multiple local IP addresses and wish to explicitly select an IP address to bind to, set the environment variable
+LOCAL_IP_ENV with the correct IP address.  For example:
+
+    sudo LOCAL_IP_ENV=192.168.0.23 make st
+
+The STs require root access to run, so they run with sudo.
+
+#### The full ST suite
+
+To run the unit tests, start by setting up a virtual env using the instructions detailed
+[here](#setting-up-a-virtualenv).
+
+From the root directory of the checked out repository, run
+
+    sudo make st
+
+The full ST suite builds the calicoctl binary and calico/node docker image.  It starts containers running etcd and
+consul, and then runs all of the tests defined in tests/st, using the built calicoctl and calico/node to run the tests.
+
+#### The fast ST suite
+
+To run the unit tests, start by setting up a virtual env using the instructions detailed
+[here](#setting-up-a-virtualenv).
+
+From the root directory of the checked out repository, run
+
+    sudo make fast-st
+
+Unlike the full ST suite, the fast ST suite does not build the calicoctl binary - instead it runs calicoctl directly
+from the python intepreter.  It starts containers running etcd and consul, and then runs all of the tests defined in
+tests/st that do not have the "slow" attrbiute assigned to them.
+
+#### Running a subset of system tests
+
+If you wish to test a single system test module (particularly useful when writing a new system test, or running a
+quiock sanity check), use the full ST make target and set the environment variable ST_TO_RUN with the path of the module
+to run, or a directory containing a set of modules to run.  For example:
+
+To run all of the BGP tests:
+
+    sudo ST_TO_RUN=calico_containers/tests/st/bgp make st
+
+To run the single no-orchestrator, mainline multi-host test:
+
+    sudo ST_TO_RUN=calico_containers/tests/st/no_orchestrator/test_mainline_multi_host.py make st
+
+
+## Setting up a virtualenv
+
+A [Python Virtual Environment](http://docs.python-guide.org/en/latest/dev/virtualenvs/) can be used for building
```

Comment:
I would amend this to read

> A Python virtual environment, *or virtualenv*,

to explain the use of the abbreviation throughout this doc.